spinalcordtoolbox/testing/create_test_data.py

```diff
@@ -151,19 +151,20 @@ def dummy_segmentation(size_arr=(256, 256, 256), pixdim=(1, 1, 1), dtype=np.floa
             if shape == 'ellipse':
                 data[:, :, iz] = (((xx - nx / 2) / radius_RL) ** 2 + ((yy - ny / 2) / radius_AP) ** 2 <= 1) * 1
     elif interleaved:
-        # define array based on a polynomial function, within Y-Z plane to simulate slicewise motion in A-P
-        y = np.matlib.repmat([round(nx / 2.) + pixdim[0]*factor, round(nx / 2.) - pixdim[0]*factor], 1, round(nz / 2))
-        if nz % 2 != 0:         # if z-dimension is odd, add one more element to fit size
-            y = numpy.append(y,round(nx / 2.) + pixdim[0]*factor)
-        y = y.reshape(nz)       # reshape to vector (1,R) -> (R,)
-        z = np.arange(0, nz)
-        p = np.poly1d(np.polyfit(z, y, deg=nz))
+        # create regularized curve, within Y-Z plane (A-P), located at x=nx/2, passing through the following points:
```

Comment:
I find it counterintuitive to have the polynomial shape only under interleaved. A clearer approach, would be to allow polynomial shape regardless what interleaved is, and in order to keep the same default behaviour, polynomial order could be set to one by default. 

also note that i am starting to see a lot of similarity with `dummy_centerline()`. At the end, a segmentation is just a centerline with a shape around it (ellipse or rectangle). Would it make sense to merge the two functions? Maybe this is a bad idea…

Reply:
>A clearer approach, would be to allow polynomial shape regardless what interleaved is, and in order to keep the same default behavior, polynomial order could be set to one by default.

Of course, it sounds better. Do we want to move to solely polynomial function (i.e., delete current code and let only code for polynomial) or let both options (polynomial and "_non-polynomial_")? So far, I modified code to choose either from original _non-polynomial_ behavior or polynomial one in this commit: https://github.com/neuropoly/spinalcordtoolbox/pull/2835/commits/6f139927e55d60765aefbd8456dec8144356cc06


> also note that i am starting to see a lot of similarity with `dummy_centerline()`. At the end, a segmentation is just a centerline with a shape around it (ellipse or rectangle). Would it make sense to merge the two functions? Maybe this is a bad idea…

Yeah, there are a lot of similarities between `dummy_centerline()` and `dummy_segmentation()`, so we could probably merge these two functions. But I am not sure, if to start with this merging within this PR, because purpose of this PR should be to create interleaved 4D data to test `sct_dmri_moco -interleaved `.

Reply:
> Do we want to move to solely polynomial function (i.e., delete current code and let only code for polynomial) or let both options (polynomial and "non-polynomial")

simplest approach is always the best. if we can achieve the same functionality (and more) with less code, we should always go with less code

Reply:
> (current non-polynomial segmentations are used in this unit testing, so probably the second option would be better)

as mentioned above ^ if we can reach the same functionality with more general code, then the unit tests should pass (if they don't pass, it means there was a change in behaviour, which we should avoid-- expect if it affects input arguments, which is manageable)

Reply:
> But I am not sure, if to start with this merging within this PR, because purpose of this PR should be to create interleaved 4D data to test sct_dmri_moco -interleaved .

right-- let's do this merge in another PR

Reply:
> as mentioned above ^ if we can reach the same functionality with more general code, then the unit tests should pass (if they don't pass, it means there was a change in behavior, which we should avoid-- expect if it affects input arguments, which is manageable)

Okay, I tried to reduce and simplify code to let only polynomial function. Unfortunately, unit tests fail (see comment below).